src/main/java/io/lettuce/core/pubsub/PubSubEndpoint.java

```diff
@@ -193,8 +194,8 @@ protected boolean containsViolatingCommands(Collection<? extends RedisCommand<?,
         return false;
     }
 
-    private static boolean isAllowed(RedisCommand<?, ?, ?> command) {
-        return ALLOWED_COMMANDS_SUBSCRIBED.contains(command.getType().name());
+    private boolean isAllowed(RedisCommand<?, ?, ?> command) {
+        return getProtocolVersion() == ProtocolVersion.RESP3 || ALLOWED_COMMANDS_SUBSCRIBED.contains(command.getType().name());
```

Comment:
It would also make sense to check against the negotiated version instead of the configured one as the default is to auto-detect the protocol version while `ClientOptions.getProtocolVersion()` returns `ProtocolVersion.newestSupported()` (RESP3). This can lead to auto-negotiation into RESP2 while `ClientOptions.getProtocolVersion()` would return `RESP3`.